### PR TITLE
build: Cleaned up the poms to match the new directory structure

### PR DESCRIPTION
**Pull request by chrisdutz**
## Description

Even if the directory structure is a lot cleaner now, still the structure of the maven reactor was flat. 
By inserting intermediate parent poms in each of the subdirectories and adjusting the parent relation, now the project should be nicely structured in Maven enabled IDEs.

I also streamlined the profiles as some were duplicates and moved the integration-test module execution into a separate "with-integration-test" profile as it's execution takes unreasonably long and should probably not be run locally.

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/executable/ExecutableManager.java [hunk 1]
```diff
@@ -227,6 +228,8 @@ protected void saveToDir(ByteBuffer byteBuffer, String destination) throws IOExc
     try {
       Path path = Paths.get(destination);
       Files.deleteIfExists(path);
+      // Ensure the target directory exists.
+      SystemFileFactory.INSTANCE.makeDirIfNecessary(destination);
```
Comment (HTHou) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/executable/ExecutableManager.java:232]: The destination String is not a Dir path, it is a file path I believe.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/file/SystemFileFactory.java [hunk 2]
```diff
@@ -64,18 +64,21 @@ public File getFile(File parent, String child) {
...
+    File directory = getFile(dir);
+    if (directory.isFile()) {
+      directory = directory.getParentFile();
+    }
+    if (directory.exists()) {
       return;
     }
-    FileUtils.forceMkdir(file);
+    FileUtils.forceMkdir(directory);
   }
```
Comment (HTHou) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/file/SystemFileFactory.java:83]: Then the non-existed file path will be made as a directory by this method.

**Comment by chrisdutz** [comment]
Unfortunately I've encountered a number of problems with the CPP builds and will be cleaning up ab bit.

**Comment by HTHou** [comment]
> Unfortunately I've encountered a number of problems with the CPP builds and will be cleaning up ab bit.

yes, I think the latest CMake has a bug on macos. It cannot find the correct homebrew path.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/executable/ExecutableManager.java (hunk 1 above)
Comment (chrisdutz) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/executable/ExecutableManager.java:232]: Yeah... I've noticed that and implemented fixes and am currently trying my changes on mac/linux/win each for x86_64 and aarch64... Just doing this in a fork in order to not spam the making list.

**Comment by chrisdutz** [comment]
Cancelling this PR and will be re-submitting one from my Fork (as this one's slightly out of sync)

**Pull request closed without merging**